File: bench/tasks/18-path-parts/zig/main.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    var gpa = std.heap.page_allocator;

    const joined = try std.fs.path.join(gpa, &[_][]const u8{ "reports", "2026", "q1.json" });
    defer gpa.free(joined);
    try stdout.print("{s}\n", .{joined});

    const path = "reports/2026/q1.json";
    try stdout.print("{s}\n", .{std.fs.path.basename(path)});
    try stdout.print("{s}\n", .{std.fs.path.dirname(path).?});
    try stdout.print("{s}\n", .{std.fs.path.extension(path)});
}

File: bench/tasks/18-path-parts/python/main.py
import os

path = os.path.join("reports", "2026", "q1.json")
print(path.replace("\\", "/"))
print(os.path.basename("reports/2026/q1.json"))
print(os.path.dirname("reports/2026/q1.json").replace("\\", "/"))
print(os.path.splitext("reports/2026/q1.json")[1])
